## 데이터 불러오기

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
df = pd.read_csv('/content/drive/MyDrive/Streamlit 해커톤/data/transactions_standard_orders.csv')
df.head(3)

ID                    _created_on               _last_updated_on  \
0  28961  2023-03-07T04:07:40.987+09:00  2023-03-07T04:07:40.987+09:00   
1  40624  2023-03-07T04:07:47.974+09:00  2023-03-07T04:07:47.974+09:00   
2   4825  2023-03-07T04:06:21.478+09:00  2023-03-07T04:06:21.478+09:00   

   _revision                          _partneruuid  _row_index  \
0          1  5c15d817-5a2c-4d77-bec4-80f5a6460f2e           0   
1          1  5c15d817-5a2c-4d77-bec4-80f5a6460f2e           0   
2          1  5c15d817-5a2c-4d77-bec4-80f5a6460f2e           0   

                   DATE_TIME  SETTLEMENT_ID               type   ORDER_ID  \
0  2022-09-10T03:05:55+09:00    16523070061  FBA Inventory Fee  220831QJO   
1  2022-11-02T19:25:58+09:00    16831770931        Service Fee        NaN   
2  2022-12-16T10:08:52+09:00    17068418651        Service Fee        NaN   

   ... OTHER_TRANSACTION_FEES OTHER   TOTAL TAX_COLLECTION_MODEL  \
0  ...                   0.00 -5.29   -5.29                  NaN   
1  ...                -375.27  0.00 -375.27                  NaN   
2  ...                -501.88  0.00 -501.88                  NaN   

  PRODUCT_SALES_TAX SHIPPING_CREDITS_TAX GIFTWRAP_CREDITS_TAX  \
0               0.0                  0.0                  0.0   
1               0.0                  0.0                  0.0   
2               0.0                  0.0                  0.0   

  PROMOTIONAL_REBATES_TAX     ACCOUNT_TYPE  MARKETPLACE_WITHHELD_TAX  
0                       0  Standard Orders                       0.0  
1                       0  Standard Orders                       0.0  
2                       0  Standard Orders                       0.0  

[3 rows x 36 columns]

In [ ]:
df.shape

(118111, 36)

* 총 11만개 행, 36개 열 존재

In [ ]:
df.columns

Index(['ID', '_created_on', '_last_updated_on', '_revision', '_partneruuid',
       '_row_index', 'DATE_TIME', 'SETTLEMENT_ID', 'type', 'ORDER_ID', 'SKU',
       'DESCRIPTION', 'QUANTITY', 'MARKETPLACE', 'FULFILLMENT', 'ORDER_CITY',
       'ORDER_STATE', 'ORDER_POSTAL', 'PRODUCT_SALES', 'SHIPPING_CREDITS',
       'GIFT_WRAP_CREDITS', 'PROMOTIONAL_REBATES', 'SALES_TAX_COLLECTED',
       'MARKETPLACE_FACILITATOR_TAX', 'SELLING_FEES', 'FBA_FEES',
       'OTHER_TRANSACTION_FEES', 'OTHER', 'TOTAL', 'ACCOUNT_TYPE',
       'MARKETPLACE_WITHHELD_TAX', 'TAX_COLLECTION_MODEL', 'PRODUCT_SALES_TAX',
       'SHIPPING_CREDITS_TAX', 'GIFTWRAP_CREDITS_TAX',
       'PROMOTIONAL_REBATES_TAX'],
      dtype='object')

* 사용할 칼럼명
  * DESCRIPTION: 상품명과 설명
  * QUANTITY: 주문 수량
  * type: 주문, 환불, 이체, 조정, FBA 재고 수수료, FBA 고객 반품 수수료

In [ ]:
df['DESCRIPTION'].head(20)

0                       FBA Removal Order: Disposal Fee
1                                   Cost of Advertising
2                                   Cost of Advertising
3       FBA Inventory Reimbursement - Damaged:Warehouse
4                       FBA Removal Order: Disposal Fee
5         FBA Inventory Reimbursement - Customer Return
6                    Ottogi Honey Citron Tea, 35 Ounces
7                       Ottogi Honey Ginger Tea, 35 oz.
8     [OTTOGI] Jin Ramen, Mild Flavor - Korean Insta...
9     [OTTOGI] Jin Ramen Spicy, Korean Style Instant...
10    [OTTOGI] Cheese Ramen, KOREAN STYLE INSTANT NO...
11    [OTTOGI] Jin Ramen, Mild Flavor - Korean Insta...
12    [OTTOGI] Jin Ramen Mild, KOREAN STYLE INSTANT ...
13    [OTTOGI] Jin Ramen Mild, KOREAN STYLE INSTANT ...
14    [OTTOGI] Jin Ramen, Spicy Flavor - Korean Inst...
15    [OTTOGI] Jin Ramen, Mild Flavor - Korean Insta...
16    [OTTOGI] Jin Ramen, Spicy Flavor - Korean Inst...
17    [OTTOGI] Jin Ramen, Spicy Flavor - Korean 

* 데이터에 이상값 존재?

In [ ]:
df[df['DESCRIPTION'].str.contains('OTTOGI')]['DESCRIPTION']

8         [OTTOGI] Jin Ramen, Mild Flavor - Korean Insta...
9         [OTTOGI] Jin Ramen Spicy, Korean Style Instant...
10        [OTTOGI] Cheese Ramen, KOREAN STYLE INSTANT NO...
11        [OTTOGI] Jin Ramen, Mild Flavor - Korean Insta...
12        [OTTOGI] Jin Ramen Mild, KOREAN STYLE INSTANT ...
                                ...                        
118106    [OTTOGI] Delicious COOKED RICE, Gluten free, M...
118107    [OTTOGI] Jin Ramen MILD - KOREAN STYLE INSTANT...
118108    [OTTOGI] Premium Roasted Sesame Oil, 100% Pure...
118109    [OTTOGI] Jin Ramen Mild, KOREAN STYLE INSTANT ...
118110    [OTTOGI] Jin Ramen MILD - KOREAN STYLE INSTANT...
Name: DESCRIPTION, Length: 112152, dtype: object

* 정상적으로 'OTTOGI'를 가지고 있는 데이터만 확인
* 118111 - 112152 = 5959개의 이상값 존재

In [ ]:
df[~df['DESCRIPTION'].str.contains('OTTOGI')]['DESCRIPTION']

0                         FBA Removal Order: Disposal Fee
1                                     Cost of Advertising
2                                     Cost of Advertising
3         FBA Inventory Reimbursement - Damaged:Warehouse
4                         FBA Removal Order: Disposal Fee
                               ...                       
117481                                Cost of Advertising
117586      FBA Inventory Reimbursement - Customer Return
117669                                Cost of Advertising
117744          FBA Amazon-Partnered Carrier Shipment Fee
117930    FBA Inventory Reimbursement - Damaged:Warehouse
Name: DESCRIPTION, Length: 5959, dtype: object

* 이상값을 확인해보니 상품명이 아닌 다른 값을 가짐 -> 이상값으로 제외

## 필요한 오뚜기 데이터 추출

In [ ]:
df_ottogi = df[df['DESCRIPTION'].str.contains('OTTOGI')]

In [ ]:
df_ottogi = df_ottogi[['DESCRIPTION', 'QUANTITY', 'type']]
df_ottogi

DESCRIPTION  QUANTITY   type
8       [OTTOGI] Jin Ramen, Mild Flavor - Korean Insta...         1  Order
9       [OTTOGI] Jin Ramen Spicy, Korean Style Instant...         1  Order
10      [OTTOGI] Cheese Ramen, KOREAN STYLE INSTANT NO...         1  Order
11      [OTTOGI] Jin Ramen, Mild Flavor - Korean Insta...         1  Order
12      [OTTOGI] Jin Ramen Mild, KOREAN STYLE INSTANT ...         1  Order
...                                                   ...       ...    ...
118106  [OTTOGI] Delicious COOKED RICE, Gluten free, M...         1  Order
118107  [OTTOGI] Jin Ramen MILD - KOREAN STYLE INSTANT...         1  Order
118108  [OTTOGI] Premium Roasted Sesame Oil, 100% Pure...         1  Order
118109  [OTTOGI] Jin Ramen Mild, KOREAN STYLE INSTANT ...         1  Order
118110  [OTTOGI] Jin Ramen MILD - KOREAN STYLE INSTANT...         1  Order

[112152 rows x 3 columns]

In [ ]:
df_ottogi['DESCRIPTION'].nunique()

112

* 총 112152개의 행, 112개의 고유한 상품 존재

In [ ]:
pd.set_option('display.max_rows', None)

In [ ]:
df_ottogi['DESCRIPTION'].value_counts()

[OTTOGI] Delicious COOKED RICE, Gluten free, Microwavable instant cooked rice, Precooked ready to eat container (7.40oz., 12 count)                       14815
[OTTOGI] Jin Ramen, Mild Flavor - Korean Instant Ramen Noodle, Best Tasting Soup Traditional Instant Ramen (120g) -18 Pack                                14811
[OTTOGI] Jin Ramen, Spicy Flavor - Korean Instant Ramen Noodle, Best Tasting Soup Traditional Instant Ramen (120g) -18 Pack                               14228
[OTTOGI] Premium Roasted Sesame Oil, 100% Pure Sesame Oil, Tradtional Korean Style oil (10.82 fl oz. 320ml)                                                7719
[OTTOGI] Jin Ramen Mild, KOREAN STYLE INSTANT CUP NOODLE, (65g) -6 Pack                                                                                    6976
[OTTOGI] JIN RAMEN KOREAN STYLE INSTANT NOODLE, RICH SPICY FLAVOR, Instant Cup Ramen, Best tasting soup ramen, traditional gourmet taste & easy to coo     6348
[OTTOGI] Cheese Ramen, KOREAN STYLE INST